src/client/opamCliMain.ml

```diff
@@ -43,25 +44,36 @@ let rec filter_cli_arg cli acc args =
     | _ ->
       filter_cli_arg cli (arg::acc) args
 
-(* Pre-process argv processing the --yes and --cli. Returns Some cli, if --cli
-   was encountered, a boolean indicating if a valid --yes/-y was encountered
-   (it returns false if multiple flags were encountered) and the list of
-   arguments to continue with processing. *)
-let rec preprocess_argv cli yes args =
+(* Pre-process argv processing the --yes, confirm-level, and --cli. Returns
+   Some cli, if --cli was encountered, a boolean indicating if a valid --yes/-y
+   was encountered (it returns false if multiple flags were encountered) and
+   the list of arguments to continue with processing. *)
+let rec preprocess_argv cli yes confirm args =
   let is_valid_yes = function [_] -> true | _ -> false in
   match args with
   | [] ->
-    (cli, is_valid_yes yes, yes)
+    (cli, is_valid_yes yes, confirm, yes)
   | "--" :: _ ->
-    (cli, is_valid_yes yes, yes @ args)
+    (cli, is_valid_yes yes, confirm, yes @ args)
   (* Note that because this is evaluated before a sub-command, all the
      prefixes of --yes are assumed to valid at all times. *)
   | ("-y" | "--y" | "--ye" | "--yes") as yes_opt :: args ->
     if yes = [] then
-      preprocess_argv cli [yes_opt] args
+      preprocess_argv cli [yes_opt] confirm args
     else
-      (cli, false, yes @ [yes_opt])
-  | "--cl" :: args -> preprocess_argv cli yes ("--cli"::args)
+      (cli, false, confirm, yes @ [yes_opt])
+  | ("--con" | "--conf" | "--confi" | "--confir" | "--confirm" | "--confirm-"
+    | "--confirm-l" | "--confirm-le" | "--confirm-lev" | "--confirm-leve"
+    | "--confirm-level") :: cl_arg :: args ->
```

Comment:
(needs fleshing out):
```suggestion
  | flag :: cl_arg :: args when String.length flag > 4 && prefix_of "--confirm-level" flag ->
```
`prefix_of s1 s2` returns `true` `s2` is a prefix of `s1`. The length `> 4` ensures that at least `--con` has been given